xarray/core/indexing.py

```diff
@@ -474,29 +460,37 @@ def _ensure_ndarray(self, value):
         return value
 
     def __getitem__(self, key):
-        key = self._convert_key(key)
         return self._ensure_ndarray(self.array[key])
 
     def __setitem__(self, key, value):
-        key = self._convert_key(key)
         self.array[key] = value
 
 
 class DaskIndexingAdapter(utils.NDArrayMixin):
-    """Wrap a dask array to support orthogonal indexing
+    """Wrap a dask array to support broadcasted-indexing.
     """
     def __init__(self, array):
         self.array = array
 
     def __getitem__(self, key):
-        key = expanded_indexer(key, self.ndim)
-        if any(not isinstance(k, integer_types + (slice,)) for k in key):
+        """ key: tuple of Variable, slice, integer """
+        # basic or orthogonal indexing
+        if all(isinstance(k, (integer_types, slice)) or k.squeeze().ndim <= 1
```

Comment:
For simple cases where everything is an integers or slice, we want to just use a single indexing call like `self.array[key]` rather than this loop.

In the hard case when some arguments are arrays, we should try `self.array.vindex[key]`. If it doesn't work in some cases, we can either add work-arounds or try to fix it upstream in dask.